code/API_definitions/network_access_management.yaml

```
@@ -0,0 +1,2863 @@
+
+    ## Optional Modes
+
+    ### Default Device
+    When an identity for OAuth 2.0 access token has a "default device", such as a typical residential customer with one service site and one network access device, service site and device scopes MAY be omitted.  Any fields referencing a network access device using the `/isolated-networks` or `/isolated-networks/*` APIs MUST be left blank by API clients.   Network operators SHOULD substitute the missing field with the default device.  API definitions include additional details for error responses so API clients can detect if this mode is available.
+
+    ## Comments for Network Operators
+    The following comments are for Network Operators implementing this API.
+
+    ### Auto-Deleting Isolated Networks
+    When an isolated network is not configured on any device the network configuration and its identifier SHOULD be auto-deleted.  API clients MUST not be expected to delete networks not configured on any device.
+
+    ### Mesh Systems
+    Mesh systems SHOULD be represented as a single device when implementing this API due to the undefined behavior of LAN connectivity across multiple devices with the same isolated network.  The recommendation is to use the gateway hardware address of the mesh system when exposing that field, though the hardware address of any node in the mesh may be used.  If an end-user replaces the equipment, all existing isolated networks SHOULD be synced to the new device representing this mesh.  Alternatively, a synthetic hardware address may also be used if the specification for the address type allows.
+
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0
+  contact:
+    email: project-manager@cablelabs.com
+
+tags:
+  - name: Service Sites
+  - name: Devices
+  - name: Isolated Networks
+  - name: Reboot Requests
+
+servers:
+  - url: "{apiRoot}/{basePath}"
+    variables:
+      apiRoot:
+        default: http://localhost:5000
+        description: API root
+      basePath:
+        default: cpe-management/v0
+        description: Base path for the CPE Management API
+
+security:
+  - oauth2: []
+
+paths:
+  /service-sites:
+    get:
+      tags:
+        - Service Sites
+      summary: Get all service sites
+      security:
+        - oauth2:
+            - cpe-management-service-sites-read
+            - cpe-management-service-sites-location-read
+            - cpe-management-devices-read

```

```suggestion
        - oAuth2ClientCredentials: []
        - threeLegged:
            - cpe-management-service-sites-read
            - cpe-management-service-sites-location-read
            - cpe-management-devices-read
```
I think that we should use three legged credentials. And this would apply to other endpoints as well.

---

@RandyLevensalor the whole security section could probably use an overhaul looking at the documented options: https://swagger.io/docs/specification/authentication/oauth2/  

The primary use case is Application Code Grant Type (3 legged), but there is note in the description that Client Credential Grant Type (2 legged) may be used in certain situations.  

> Authorization of API clients SHOULD be performed with OAuth 2.0 using the authorization code grant type such that service owners delegate access to their resources exposed by this API to the API client.  PKCE MUST be used when the API client secret cannot be secured (e.g., mobile application or single page application without a secure back-end).  This API MAY also be used with the client credentials grant type, though such a use case only makes sense for first-party network operator API clients or when the service owner operates their own API client (client ID is bound to the service owner's identity).

---

Migrated to openId per https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#use-of-openidconnect-for-securityschemes

I still have a few endpoints to touch up before resolving.

---

Finished adding sec defs to endpoints.